frontend/javascripts/oxalis/model/bucket_data_handling/wkstore_adapter.js

```diff
@@ -145,23 +146,41 @@ export async function requestFromStore(
     createRequestBucketInfo(zoomedAddress, resolutions, fourBit, applyAgglomerates, version),
   );
 
-  return doWithToken(async token => {
-    const { buffer: responseBuffer, headers } = await Request.sendJSONReceiveArraybufferWithHeaders(
-      `${dataUrl}/data?token=${token}`,
-      {
+  try {
+    return await doWithToken(async token => {
```

Comment:
Finally found why the errors were not caught: The `await` statement was missing.
@philippotto Could you please have another look at the code again? I applied your feedback and all errors should now be caught properly.

Reply:
Hmm, not sure I understand this fix. `doWithToken` returns a promise and there shouldn't be a difference between simply returning this promise or awaiting that promise -- thus creating a new promise -- and returning that new promise.
See the highlighted section here for another example: https://javascript.info/async-await#:~:text=We%20could%20explicitly%20return%20a%20promise,%20which%20would%20be%20the%20same:

Or am I missing something?

Reply:
Oh, there wasn't a bug in the beginning but I added one temporarily:
 If the await statement is missing, the promise of `doWithToken` is directly returned and the catch statement will never catch a failing request as the promise is already returned. But if I wait for `doWithToken` to succeed or fail, the catch block correctly catches the error :+1:

Reply:
Ahh, thanks for the clarification :+1: I didn't consider the try-catch-block in this context. Makes total sense :+1: